### PR TITLE
Add Connection Validation and Pool configuration

### PR DESCRIPTION
**Pull request by tnleeuw**

### CODE REVIEW

File: core/src/main/java/org/frankframework/jta/narayana/NarayanaDataSourceFactory.java [hunk 1]
```diff
@@ -15,104 +15,59 @@
...
 
 	public NarayanaDataSourceFactory() {
+		// For backwards compatibility, apply these configuration constants if they're found.
 		AppConstants appConstants = AppConstants.getInstance();
 		minPoolSize = appConstants.getInt("transactionmanager.narayana.jdbc.connection.minPoolSize", minPoolSize);
 		maxPoolSize = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxPoolSize", maxPoolSize);
-		maxLifeTimeSeconds = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxLifeTime", maxLifeTimeSeconds);
+		maxIdle = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxIdle", maxIdle);
+		maxLifeTime = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxLifeTime", maxLifeTime);
+		testQuery = appConstants.getString("transactionmanager.narayana.jdbc.connection.testQuery", testQuery);
```
Comment (jkosternl) [inline core/src/main/java/org/frankframework/jta/narayana/NarayanaDataSourceFactory.java:50]: All these extra/new features we set are valuable improvements, to my humble opinion, even though perhaps Ali's main issue was solved differently.

File: core/src/main/java/org/frankframework/receivers/Receiver.java [hunk 2]
```diff
@@ -1290,6 +1290,9 @@ private Message processMessageInAdapter(MessageWrapper<M> messageWrapper, PipeLi
 						}
 						getListener().afterMessageProcessed(pipeLineResult, messageForAfterMessageProcessed, session);
 					} catch (Exception e) {
+						itx.setRollbackOnly();
+						messageInError = true;
+						pipeLineResult.setState(ExitState.ERROR);
```
Comment (jkosternl) [inline core/src/main/java/org/frankframework/receivers/Receiver.java:1295]: Woops!!? This is from 'my' issue!

File: test/src/main/configurations/JMS/ConfigurationEsbJmsListenerSender.xml [hunk 3]
```diff
@@ -308,13 +308,14 @@
...
 			className="nl.nn.adapterframework.errormessageformatters.SoapErrorMessage" />
-		<receiver>
+		<receiver maxDeliveries="2">
 			<listener className="nl.nn.adapterframework.extensions.esb.EsbJmsListener"
 		  		destinationName="${jms.destination.i4testiaf_in}" jmsRealm="qcf"
 				messageProtocol="RR"
 				messageSelector="SOURCE='FFTest_esbJmsRR_2'"
 				xPathLoggingKeys="value"
 			/>
+			<errorStorage className="org.frankframework.jdbc.JdbcTransactionalStorage" slotId="${applicationId}/Shit" />
```
Comment (jkosternl) [inline test/src/main/configurations/JMS/ConfigurationEsbJmsListenerSender.xml:318]: Woops also for line 311?!

**Review by jkosternl — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/frankframework/receivers/Receiver.java (hunk 2 above)
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/receivers/Receiver.java:1295]: Oops you're right. Another thing that leaked in 😳 Forgot to clean the workspace before doing the ports.

**Review by nielsm5 — APPROVED** [review]

File: core/src/main/resources/springTOMCAT.xml [hunk 4]
```diff
@@ -30,6 +30,7 @@
 		</property>
 	</bean>
 
+	<bean id="dataSourceFactory" class="nl.nn.adapterframework.jndi.PoolingJndiDataSourceFactory" />
```
Comment (nielsm5) [inline core/src/main/resources/springTOMCAT.xml:33]: Oei!

**Pull request merged**